synth_node_bin/src/action/mod.rs

```diff
@@ -15,42 +16,72 @@ trait SynthNodeAction {
     /// It can be displayed during the runtime.
     fn info(&self) -> &str;
 
+    /// Action configuration.
+    ///
+    /// Allows preconfiguration settings before the action execution starts.
+    fn config(&self) -> ActionCfg;
+
     /// Defines the core action functionality.
     ///
     /// All the program logic happens here.
     async fn run(&self, synth_node: &mut SyntheticNode, addr: SocketAddr) -> Result<()>;
 }
 
 /// List of available actions.
+// TODO: Make a command argument to choose action type.
+#[allow(dead_code)]
 pub enum ActionType {
     SendGetAddrAndForeverSleep,
+    // TODO(Rqnsom): Add support for choosing listening address in config and apply it in the main.rs here. Details:
```

Comment:
What are the real-life scenarios where we'd like to narrow listening to some specified IP instead of 0.0.0.0 (all) for synth node binary?
Is it worth time spent?

Reply:
Yes. But because the default choice in our code is LOCALHOST for the test suite, and the synth node binary mostly uses 0.0.0.0 ----> so we should be able to change it automatically, not doing manual code edits